video/out/vo.c

```diff
@@ -1171,6 +1171,16 @@ void vo_redraw(struct vo *vo)
     mp_mutex_unlock(&in->lock);
 }
 
+// Same as vo_redraw but the redraw is delayed until it
+// is detected in the playloop.
+void vo_set_want_redraw(struct vo *vo)
+{
+    struct vo_internal *in = vo->in;
+    mp_mutex_lock(&in->lock);
+    in->want_redraw = true;
```

Comment:
```
    bool request_redraw;            // redraw request from player to VO
    bool want_redraw;               // redraw request from VO to player
```

So we ask player for redraw. I don't know what is correct approach in this case, but might need `wakeup_core(vo);` here, else playloop might be sleeping and never react to that flag.

Reply:
Also isn't `VOCTRL_UPDATE_RENDER_OPTS` doing similar things?

Reply:
Good observation, but I think it should be `mp_wakeup_core` in player/command. The approach here is to wait for the playloop in the main thread to read this. Then that does `vo_redraw` which will wakeup the vo thread if needed.